node/node_test.go

```diff
@@ -135,6 +136,25 @@ func TestNodeSetAppVersion(t *testing.T) {
 	assert.Equal(t, n.nodeInfo.(p2p.DefaultNodeInfo).ProtocolVersion.App, appVersion)
 }
 
+func TestPprofServer(t *testing.T) {
+	config := test.ResetTestRoot("node_pprof_test")
+	defer os.RemoveAll(config.RootDir)
+	config.RPC.PprofListenAddress = testFreeAddr(t)
+
+	// should not work yet
+	_, err := http.Get("http://" + config.RPC.PprofListenAddress)
+	assert.Error(t, err)
+
+	n, err := DefaultNewNode(config, log.TestingLogger())
+	assert.NoError(t, err)
+	assert.NoError(t, n.Start())
+	defer n.Stop()
+	assert.NotNil(t, n.pprofSrv)
+
+	_, err = http.Get("http://" + config.RPC.PprofListenAddress)
```

Comment:
This just tests that there is _a_ server running, not that it's serving the pprof data. I'd add a check that the response doesn't return `404` on `/debug/pprof/`.

Reply:
good point